course_info/tests/jasmine/controllers/PeopleControllerSpec.js

```diff
@@ -74,61 +56,632 @@ describe('Unit testing PeopleController', function() {
+
+                $httpBackend.expectPOST(coursePeopleURL, user)
+                    .respond(500, testFailureData);
+                $httpBackend.flush(1);
+
+                expect(scope.messages.warnings).toEqual([expectedWarning]);
+                expect(ajaxResponse.data).toEqual(testFailureData);
+            });
+            it('tracks partial failure, logs appropriate error message, and ' +
+                'does not reject promise chain', function() {
+                var testFailureData = {
+                    detail: 'Canvas API error details: testing'};
+                var expectedWarning = {
+                    type: 'partialFailure',
+                    failureDetail: testFailureData.detail,
+                    name: args.name,
+                    searchTerm: args.searchTerm
+                };
+
+                $httpBackend.expectPOST(coursePeopleURL, user)
+                    .respond(500, testFailureData);
+                $httpBackend.flush(1);
+
+                expect(scope.messages.warnings).toEqual([expectedWarning]);
+                expect(ajaxResponse.data).toEqual(testFailureData);
+            });
+        });
+
     });
     describe('confirmAddPeopleToCourse', function() {
-        it('reflects the right role and number of people');
-        it('initiates the process if confirmed');
-        it('doesn\'t initiate the process if canceled');
+        beforeEach(function() {
+            controller = $controller('PeopleController', {$scope: scope});
+            clearInitialCourseInstanceFetch();
+            var searchTerm ="123456789, \n987654321";
+            spyOn(scope, 'addPeopleToCourse');
+            scope.confirmAddPeopleToCourse(searchTerm);
+            scope.$digest();  // resolves modal instantiation
+        });
+
+        it('reflects the right role and number of people', function() {
+            // NOTE: currently, the modal scope shows up as a sibling of our
+            // application scope, so we're only looking there.  if that changes,
+            // this logic may need to be extended to walk the tree from
+            // $rootScope.
+            var modalScope;
+            for (modalScope = scope.$$nextSibling;
+                    modalScope != null; modalScope = modalScope.$$nextSibling) {
+                if (modalScope.hasOwnProperty('numPeople')  &&
+                        modalScope.hasOwnProperty('selectedRoleName')) {
+                    break;
+                }
+            }
+
+            // if modalScope is null, we couldn't find the modal's scope
+            expect(modalScope).not.toBeNull();
+
+            expect(modalScope.numPeople).toEqual(2); // 2 items in searchTerm
+            // this is kinda tautological, since the modal scope gets its
+            // value from scope.selectedRole.roleName, but doesn't hurt.
+            expect(modalScope.selectedRoleName)
+                .toEqual(scope.selectedRole.roleName);
```

Comment:
Should we set the `scope.selectedRole` somewhere first, so we're sure it propagates? Otherwise this probably just checks an `undefined` case.